seacrowd/sea_datasets/thai_ser/thai_ser.py

```diff
@@ -0,0 +1,200 @@
+            schema=f"seacrowd_{SEACROWD_SCHEMA_NAME}",
+            subset_id=f"{_DATASETNAME}_zoom",
+        ),
+        # studio
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_studio_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=f"{_DATASETNAME}_studio",
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_studio_seacrowd_{SEACROWD_SCHEMA_NAME}",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema=f"seacrowd_{SEACROWD_SCHEMA_NAME}",
+            subset_id=f"{_DATASETNAME}_studio",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source" or self.config.schema == f"seacrowd_{self.SEACROWD_SCHEMA_NAME}":
+            # same as schemas.speech_features(self._LABELS) except for sampling_rate
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "path": datasets.Value("string"),
+                    "audio": datasets.Audio(sampling_rate=44_100),
+                    "speaker_id": datasets.Value("string"),
+                    "labels": datasets.ClassLabel(names=self._LABELS),
+                    "metadata": {
+                        "speaker_age": datasets.Value("int64"),
+                        "speaker_gender": datasets.Value("string"),
+                    },
+                }
+            )
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        setting = "studio" if "studio" in self.config.name else "zoom"
+
+        data_paths = {"actor_demography": Path(dl_manager.download_and_extract(_URLS["actor_demography"])), "emotion_label": Path(dl_manager.download_and_extract(_URLS["emotion_label"])), setting: {}}
+        for url_name, url_path in _URLS[setting].items():
+            data_paths[setting][url_name] = Path(dl_manager.download_and_extract(url_path))
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "actor_demography_filepath": data_paths["actor_demography"],
+                    "emotion_label_filepath": data_paths["emotion_label"],
+                    "data_filepath": data_paths[setting],
+                    "split": "train",
+                },
+            )
+        ]
+
+    def _generate_examples(self, actor_demography_filepath: Path, emotion_label_filepath: Path, data_filepath: Path, split: str) -> Tuple[int, Dict]:
```

Comment:
```suggestion
    def _generate_examples(self, actor_demography_filepath: Path, emotion_label_filepath: Path, data_filepath: Dict[str, Union[Path, Dict]], split: str) -> Tuple[int, Dict]:
```
The type-hint was wrong. This suggestion gives a more explicit (and more concise) declaration.
Please remember to import the `typing.Union` if the suggestion is adopted.